design/ErrorHandling.md

```diff
@@ -23,6 +23,146 @@ There are several types of WebGPU calls that get their errors handled differentl
 Implementations should provide a way to enable synchronous validation, for example via a debug shim or via the developer tools.
 The extra overhead needs to be low enough that applications can still run while being debugged.
 
+## *Fatal Errors*: Lost/Recovered Events
+
+<!-- calling this revision 6 -->
+
+```webidl
+interface GPUDeviceLostInfo {
+    readonly attribute DOMString message;
+};
+
+partial interface GPUDevice {
+    readonly attribute Promise<GPUDeviceLostInfo> lost;
+};
+```
+
+`GPUAdapter.requestDevice` requests a device from the adapter.
+It returns a Promise which resolves when a device is ready.
+The Promise may not resolve for a long time - it resolves when the browser is ready for the application to bring up (or restore) its content.
+If the adapter is lost before the Promise resolves, the Promise rejects.
```

Comment:
The adapter. If the adapter is lost, it can't create a device. I'll rephrase it to express the intent though.

> If the adapter is unable to create a device (i.e. because the adapter was lost), the Promise rejects.